docs/overview/hooks.md

```diff
@@ -14,60 +14,84 @@ title: Lifecycle hooks overview
+
+For example, you can wrap an existing model in a snapshot processor which transforms a snapshot from the server into the shape your model expects with `preProcess`:
+
+```javascript
+const TodoModel = types.model("Todo", { done: true });
+
+const Todo = types.snapshotProcessor(TodoModel, {
+  preProcess(snapshot) {
+    return {
+        // auto convert strings to booleans as part of preprocessing
+        done: snapshot.done === "true" ? true : snapshot.done === "false" ? false : snapshot.done
+    }
+});
+
+const todo = Todo.create({ done: "true" }) // snapshot will be transformed on the way in
+```
+
+Snapshots can also be transformed from the base shape generated by `mobx-quick-tree` using the `postProcess` hook. For example, we could add a timestamp for when the snapshot was generated to the snapshot:
+
+```javascript
+const TodoModel = types.model("Todo", { done: true });
+
+const Todo = types.snapshotProcessor(TodoModel, {
+  postProcess(snapshot, node) {
+    return {
+        ...snapshot,
+        // add an extra property
+        generatedAt: new Date().getTime()
+    }
+});
+
+const todo = Todo.create({done: true});
+const snapshot = getSnapshot(todo); 
+// { done: true, generatedAt: 1699504649386 }
+```
+
+| Hook                        | Meaning                                                                                                                                                                                                                                                                                                                                                                |
+| --------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `preProcess(inputSnapshot)` | Transform a snapshot before it is applied to a node. The output snapshot must be valid for application to the wrapped type. The `preProcess` hook is passed the input snapshot, but not passed the node, as it is not done being constructed yet, and not attached to the tree. If you need to modify the node in the context of the tree, use the `afterCreate` hook. |
+| `postProcess(outputSnapshot, node)` | Transform a snapshot after it has been generated from a node. The transformed value will be returned by `getSnapshot`. The `postProcess` hook is passed the initial outputSnapshot, as well as the instance object the snapshot has been generated for. It is safe to access properties of the node or other nodes when post processing snapshots. |
```

Comment:
**praise:** @airhorns thank you. If anyone else is looking at this PR as opposed to #2065, I think this extra documentation makes the case for why this may be a balanced approach.